src/fidesops/api/v1/endpoints/oauth_endpoints.py

```diff
@@ -157,3 +168,34 @@ def read_scopes() -> List[str]:
     """Returns a list of all scopes available for assignment in the system"""
     logging.info("Getting all available scopes")
     return SCOPE_REGISTRY
+
+
+@router.post(OAUTH_CALLBACK, response_model=None)
+def oauth_callback(code: str, state: str, db: Session = Depends(get_db)):
+    """
+    Uses the passed in code to generate the token access request
+    for the connection associated with the given state.
+    """
+
+    # find authentication request by state
+    authentication_request: AuthenticationRequest = AuthenticationRequest.get_by(
```

Comment:
authenitcationrequest is a new table used to associate an authentication request `state` to a SaaS connector. When we receive the callback to this endpoint, we use the state to lookup the correct SaaS connection config to update the access token for.